src/main.c

```diff
@@ -1222,9 +1222,56 @@ static int on_config_max_connections(h2o_configurator_command_t *cmd, h2o_config
 
 static int on_config_num_threads(h2o_configurator_command_t *cmd, h2o_configurator_context_t *ctx, yoml_t *node)
 {
-    if (h2o_configurator_scanf(cmd, node, "%zu", &conf.num_threads) != 0)
+    size_t num_threads = 0;
```

Comment:
Could you please move the variable declaration inside `if (node->type == YOML_TYPE_SCALAR) {`? I think that the only block that uses the variable.